File: src/FrameIO/Models/ComPortTypeModels.cs
using FrameIO.Data;
using System;
using static FrameHelper.EntityHelpers;

namespace FrameIO.Models
{
    public interface IComPortTypeDTO : IWithIdAndTimeStamp
    {
        PortType PortType { get; set; }
        string AddressFormat { get; set; }
        string ReadProtocol { get; set; }
        string WriteProtocol { get; set; }
    }

    public class ComPortTypeDTO : IComPortTypeDTO
    {
        public Guid Id { get; set; }
        public DateTime TimeStamp { get; set; }
        public PortType PortType { get; set; }
        public string AddressFormat { get; set; }
        public string ReadProtocol { get; set; }
        public string WriteProtocol { get; set; }

    }

    public interface IComPortTypeView : IComPortTypeDTO
    {
    }

    public class ComPortTypeView : ComPortTypeDTO, IComPortTypeView
    {
    }

}

File: src/FrameUI/src/app/models/ComPortTypeModels.ts

import { PortType } from './ComConfigModels';

export interface IcomPortTypeDTO {
    id: string;
    timeStamp: Date;
    portType: PortType;
    addressFormat: string;
    readProtocol: string;
    writeProtocol: string;
    
}

export class ComPortTypeDTO implements IcomPortTypeDTO {
    public id: string;
    
    public timeStamp: Date;
    
    public portType: PortType;
    
    public addressFormat: string;
    
    public readProtocol: string;
    
    public writeProtocol: string;
    
    constructor(model?: IcomPortTypeDTO) {
        if(model) {
            this.id = model.id;
            this.timeStamp = model.timeStamp;
            this.portType = model.portType;
            this.addressFormat = model.addressFormat;
            this.readProtocol = model.readProtocol;
            this.writeProtocol = model.writeProtocol;
            
        }
    }
}

export interface IcomPortTypeView extends IcomPortTypeDTO {
    
}

export class ComPortTypeView extends ComPortTypeDTO implements IcomPortTypeView {
    constructor(model?: IcomPortTypeView) {
	super(model);

        if(model) {
            
        }
    }
}
 